public/app/features/integrated-alerting/components/AlertRules/AlertRulesTable/AlertRulesTable.tsx

```diff
@@ -1,78 +1,141 @@
+    prepareRow,
+    pageCount,
+    setPageSize,
+    gotoPage,
+    state: { pageSize, pageIndex },
+  } = tableInstance;
+
+  const onPageChanged = (newPageIndex: number) => {
+    gotoPage(newPageIndex);
+    onPaginationChanged(pageSize, newPageIndex);
+  };
+
+  const onPageSizeChanged = (newPageSize: number) => {
+    setPageSize(newPageSize);
+    onPaginationChanged(newPageSize, pageIndex);
+  };
 
   return (
-    <div className={style.tableWrap} data-qa="alert-rules-table-outer-wrapper">
-      <div className={style.table} data-qa="alert-rules-inner-wrapper">
-        {pendingRequest ? (
-          <div data-qa="alert-rules-table-loading" className={style.empty}>
-            <Spinner />
-          </div>
-        ) : null}
-        {!rows.length && !pendingRequest ? (
-          <div data-qa="alert-rules-table-no-data" className={style.empty}>
-            {<h1>{emptyMessage}</h1>}
-          </div>
-        ) : null}
-        {rows.length && !pendingRequest ? (
-          <table {...getTableProps()} data-qa="alert-rules-table">
-            <thead data-qa="alert-rules-table-thead">
-              {headerGroups.map(headerGroup => (
-                <tr {...headerGroup.getHeaderGroupProps()}>
-                  {headerGroup.headers.map(column => (
-                    <th
-                      className={css`
-                        cursor: pointer;
-                        width: ${column.width};
-                      `}
-                      {...column.getHeaderProps()}
-                    >
-                      {column.render('Header')}
-                    </th>
-                  ))}
-                </tr>
-              ))}
-            </thead>
-            <tbody {...getTableBodyProps()} data-qa="alert-rules-table-tbody">
-              {rows.map(row => {
-                prepareRow(row);
-                const alertRule = row.original as AlertRule;
+    <>
+      <div className={style.tableWrap} data-qa="alert-rules-table-outer-wrapper">
+        <div className={style.table} data-qa="alert-rules-inner-wrapper">
+          {pendingRequest ? (
+            <div data-qa="alert-rules-table-loading" className={style.empty}>
+              <Spinner />
+            </div>
+          ) : null}
+          {!pageCount && !pendingRequest ? (
+            <div data-qa="alert-rules-table-no-data" className={style.empty}>
+              {<h1>{emptyMessage}</h1>}
+            </div>
+          ) : null}
+          {pageCount && !pendingRequest ? (
+            <table {...getTableProps()} data-qa="alert-rules-table">
+              <thead data-qa="alert-rules-table-thead">
+                {headerGroups.map(headerGroup => (
+                  <tr {...headerGroup.getHeaderGroupProps()}>
+                    {headerGroup.headers.map(column => (
+                      <th
+                        className={css`
+                          cursor: pointer;
+                          width: ${column.width};
+                        `}
```

Comment:
This should go in a `styles` file

Reply:
I haven't change this, I'll just put that in the table refactor